src/PowerShellEditorServices.Protocol/Server/DebugAdapter.cs

```diff
@@ -289,18 +276,28 @@ protected void Stop()
                 }
             }
 
-            if (string.IsNullOrEmpty(workingDir))
+            // A null or empty string value for CWD is an indicator to NOT change the working directory
+            // but only if we are debugging in the integrated console where there is an existing cwd.
+            // However, if we are running in a temporary integrated console, there is no pre-existing
+            // cwd, so fall through to the "else if: where we set a working directory.
+            if (string.IsNullOrEmpty(workingDir) && !launchParams.CreateTemporaryIntegratedConsole)
             {
+                Logger.Write(LogLevel.Verbose, "Launch config requested working dir not be changed/set");
```

Comment:
Doesn't look like this if is doing anything besides logging... maybe refactor to just an if statement and then after the if log the location of the workingDir?

Reply:
Well, in the case above we don't know or care about the location of the working dir but we can simplify the logic and put in a single log call.  Inside that log call I can use a ternary to determine what to log.